encrypt.go

```diff
@@ -139,47 +138,45 @@ func (es *encryptStream) encryptBlock(isFinal bool) error {
 	return nil
 }
 
-// Do some sanity checking on the receivers. Check that receivers
-// aren't sent to twice; check that there's at least one receiver.
-func (es *encryptStream) checkReceivers(v []BoxPublicKey) error {
+func checkKnownVersion(version Version) error {
+	for _, knownVersion := range KnownVersions() {
+		if version == knownVersion {
+			return nil
+		}
+	}
+	return ErrBadVersion{version}
+}
 
-	if len(v) == 0 {
+// maxEncryptReceiverCount is the maximum number of receivers allowed
+// for a single encrypted saltpack message. It is the maximum length
+// of a msgpack array, less the number of fields before the recipient
+// list.
+const maxEncryptReceiverCount = 0x7ffffff - 5
+
+// checkEncryptReceivers does some sanity checking on the
+// receivers. Check that receivers aren't sent to twice; check that
+// there's at least one receiver and not too many receivers.
+func checkEncryptReceivers(receivers []BoxPublicKey) error {
+	if len(receivers) <= 0 || len(receivers) > maxEncryptReceiverCount {
 		return ErrBadReceivers
 	}
 
-	tot := uint64(0)
-
 	// Make sure that each receiver only shows up in the set once.
-	receiversAsSet := make(map[string]struct{})
+	receiverSet := make(map[string]bool)
 
-	for _, receiver := range v {
-		// Make sure this key hasn't been used before
+	// Make sure each key hasn't been used before.
+	for _, receiver := range receivers {
 		kid := receiver.ToKID()
-		kidString := hex.EncodeToString(kid)
-		if _, found := receiversAsSet[kidString]; found {
+		kidString := string(kid)
```

Comment:
cleaned this up a bit -- idiomatic go set is `map[...]bool`, and no need to hex-encode to turn into a map key

Reply:
Is the idiomatic go set `map[...]interface{}`, to avoid storing the bool? But maybe that's more inconvenient to query, because you need to use `_, exists := ...`? I don't care either way.

Reply:
I actually agree with you, but `map[...]bool` seems more idiomatic to me given that it's in "Effective Go": https://golang.org/doc/effective_go.html#maps